Text_Mining.py

```diff
@@ -0,0 +1,161 @@
+import requests
+import pickle
+import os.path
+import nltk
+
+
+"""Downloads 'Alice in Wonderland'
+"""
+
+# Save data to a file (will be part of your data fetching script)
+if os.path.isfile('alice_in_wonderland.pickle') == False:
+    alice_in_wonderland_full_text = requests.get('http://www.gutenberg.org/files/11/11-0.txt').text
+    f = open('alice_in_wonderland.pickle', 'wb')
+    pickle.dump(alice_in_wonderland_full_text, f)
+    f.close()
+else:
+    # Load data from a file (will be part of your data processing script)
+    input_file = open('alice_in_wonderland.pickle', 'rb')
+    reloaded_copy_of_texts1 = pickle.load(input_file)
+
+
+"""Downloads 'Tom Sawyer'"""
+
+if os.path.isfile('tom_sawyer.pickle') == False:
+    Tom_Sawyer_full_text = requests.get('http://www.gutenberg.org/files/74/74-0.txt').text
+    f2 = open('tom_sawyer.pickle', 'wb')
+    pickle.dump(Tom_Sawyer_full_text, f)
+    f.close()
+else:
+    # Load data from a file (will be part of your data processing script)
+    input_file2 = open('tom_sawyer.pickle', 'rb')
+    reloaded_copy_of_texts2 = pickle.load(input_file2)
+
+"""Cuts out beginning and end sections from each book"""
+
+def cut_alice(text):
+    index = text.find('***')
+    new_text = text[index+3:]
+    index2 = new_text.find('***')
+    final_text = new_text[index2+3:]
+    index3 = final_text.find('THE END')
+    text3 = final_text[:index3+7]
+    return text3
+
+
+def cut_tom(text):
+    index = text.find('Widger')
+    final_text = text[index+6:]
+    index2 = final_text.find('End of the Project')
+    text3 = final_text[:index2-1]
+    return text3
+
+
+def parse_text(text):
+    """Cuts the beginning and end off book to get just the actual text
+    and removes spaces/punctuation. Also, changes all words to lowercase. Outputs
+    a histogram with each word and the number of times it appears in the book.
+    """
+    tokens = nltk.word_tokenize(text)
+    myDictionary = dict()
+    for word in tokens:
+        if word == "." or word == "," or word == "--" or word == ";" or word == ")" or word == "(" or word == "?" or word == "'" or word == "!" or word == ":" or word == "&":
+            tokens.remove(word)
+        elif word[0] == '_' and word[len(word)-1] == '_':
```

Comment:
Very complex chain of conditionals here. Comments explaining what this does would be good.